src/main.ts

```diff
@@ -65,27 +72,31 @@ export async function run(inputs: RunInputs): Promise<void> {
     const outputPath = path.join(tempDir(), ATTESTATION_FILE_NAME)
     core.setOutput('bundle-path', outputPath)
 
-    const att = await createAttestation(subjects, predicate, {
+    const opts: CreateAttestationOptions = {
       sigstoreInstance,
       pushToRegistry: inputs.pushToRegistry,
       githubToken: inputs.githubToken
-    })
+    }
 
-    logAttestation(subjects, att, sigstoreInstance)
+    let atts: AttestResult[]
+    if (inputs.singleSubjectAttestations) {
+      atts = await createSingleSubjectAttestations(subjects, predicate, opts)
```

Comment:
Minor stylistic note: the `createSingle/Multi` funcs are basically one liners…

I think I'd prefer to see:
- these funcs removed, so that the `createAttestation` calls happen directly in this if statement, so that the calls happen in the main func here, OR
- modifying `createAttestation` to accept an option

but I'm leaning more towards the first one.

Reply:
I've inlined the relevant functions (I have no real experience with TypeScript, so I'm not sure what the dominant style is w.r.t. factoring out functions etc.)